src/SystemMetadata.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Storj\Uplink;
+
+use DateTimeImmutable;
+use FFI\CData;
+
+class SystemMetadata
+{
+    private DateTimeImmutable $created;
+
+    private ?DateTimeImmutable $expires;
+
+    private int $contentLength;
+
+    public function __construct(DateTimeImmutable $created, ?DateTimeImmutable $expires, int $contentLength)
```

Comment:
I don't believe this class will be ever created by the client. It should be only returned by the system. Perhaps we don't need this constructor.

Reply:
I have changed the constructor to create the object from the C struct.

Also for classes ObjectInfo and BucketInfo